moban/mobanfile/targets.py

```diff
@@ -34,11 +34,16 @@ def _handle_explicit_target(options, target):
     data_file = target.get(constants.LABEL_CONFIG, common_data_file)
     output = target[constants.LABEL_OUTPUT]
     template_type = target.get(constants.LABEL_TEMPLATE_TYPE)
+    needs_ad_hoc = False
+    if template_type and len(template_type) > 0:
```

Comment:
In here, what if we do temporary template registration here? since ENGINES is a global thingy and in our hand.

import uuid

adhoc_type = uuid.uuid4.hex()
the_adhoc_type = { adhoc_type: {...whatever the ad hoc type definiotion is ...}}
plugins.ENGINES.register_options(the_adhoc_type)

Then this ad_hoc template type will get registered already., then we can set this template target with template to `adhoc_type`.

Reply:
With above modification, you may find something interesting thing happening in mobanfile/__init__.py where no changes is required :)

Reply:
wait, this adhoc template type would have associate with the template target via file extensions.

Reply:
had a look again,

import uuid

adhoc_type = uuid.uuid4.hex()
file_extension = uuid.uuid4.hex()
the_adhoc_type = { adhoc_type: {"file_extensions":[file_extension],...whatever the ad hoc type definiotion is ...}}
plugins.ENGINES.register_options(the_adhoc_type)

template_target = TemplateTarget(...template_type=file_extension) # <---

Then, the rest should work along with user defined template types

Reply:
What do you think about this suggestion?

Reply:
Looks good. :+1: